app/models/host/managed.rb

```diff
@@ -537,11 +537,29 @@ def jumpstart?
     operatingsystem.family == "Solaris" and architecture.name =~/Sparc/i rescue false
   end
 
+  HOSTGROUP_INHERITED_ATTRIBUTES = %w{puppet_proxy_id puppet_ca_proxy_id environment_id compute_profile_id}
+
+  def apply_inherited_attributes(attributes)
+    new_hostgroup_id = attributes['hostgroup_id']
+    return attributes if new_hostgroup_id.nil?
+
+    hostgroup = self.hostgroup || Hostgroup.find(new_hostgroup_id)
+    hostgroup = Hostgroup.find(new_hostgroup_id) unless hostgroup.id == new_hostgroup_id
+    return attributes unless hostgroup
```

Comment:
I don't want to change the active record yet, it will be changed later. The problem is I do need the new hostgroup object here.